.github/actions/setup-self-hosted-action/action.yml

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: 'Setup environment for self-hosted runners'
+description: 'Setup action to run jobs in a self-hosted runner'
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install python 3.7
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.7"
+    - name: Install python 3.8
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.8"
+    - name: Install python 3.9
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.9"
+    - name: Set Java Version
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: 8
+    - name: Set Go Version
+      uses: actions/setup-go@v3
+      with:
+        go-version: '>=1.16.0'
```

Comment:
This version is probably not right. Go SDK tests require >=1.18, and playground tests require 1.16.

I would recommend installing both separately.

Reply:
Thanks for the feedback Danny!
As for the Go version, since this will only be used for precommit and postcommit tasks, would it be ok to only install >=1.18?
The way we thought about this was to install the bare minimum and let each workflow install its appropriate dependencies.

Reply:
That should be fine, I would pin it to 1.18 though (not >=1.18) since right now the Go SDK guarantees compat back to 1.18

Reply:
sounds like a plan! Thanks Danny!